### PR TITLE
Enable eslint

### PR DESCRIPTION
**Pull request by dmlap**
Add an eslint config loosely based on the video.js standard one. Remove jshint. Fix up a number of style issues.

### CODE REVIEW

File: test/utils/multi-channel-captions.js [hunk 1]
```diff
@@ -1,996 +1,996 @@
 module.exports = [
-  {"type":1,"pts":7192231956,"ccData":36782},
```
Comment (dmlap) [inline test/utils/multi-channel-captions.js]: The changes in this file are strictly whitespace.

File: lib/m2ts/caption-stream.js [hunk 2]
```diff
@@ -238,7 +237,7 @@ var PADDING                    = 0x0000,
     ROLL_UP_2_ROWS             = 0x1425,
     ROLL_UP_3_ROWS             = 0x1426,
     ROLL_UP_4_ROWS             = 0x1427,
-    RESUME_DIRECT_CAPTIONING   = 0x1429,
+    // RESUME_DIRECT_CAPTIONING   = 0x1429,
```
Comment (gesinger) [inline lib/m2ts/caption-stream.js]: Is it worth keeping around if the list isn't exhaustive?

File: test/transmuxer.test.js [hunk 3]
```diff
@@ -2,63 +2,57 @@
...
+    validateTrackFragment,
+
+    pesHeader,
+    transportPacket,
+    videoPes,
+    adtsFrame,
+    audioPes,
+    timedMetadataPes;
+
+mp4.tools = require('../lib/tools/mp4-inspector');
```
Comment (gesinger) [inline test/transmuxer.test.js]: Should this be moved up to the other requires?

File: test/transmuxer.test.js [hunk 4]
```diff
@@ -93,7 +87,7 @@ QUnit.test('buffers partial packets', function() {
   var
     datas = [],
     partialPacket1 = new Uint8Array(187),
-    partialPacket2 =  new Uint8Array(189);
+  partialPacket2 = new Uint8Array(189);
```
Comment (gesinger) [inline test/transmuxer.test.js]: Any reason for this move to a new line?

File: test/transmuxer.test.js [hunk 5]
```diff
@@ -2518,9 +2512,9 @@ QUnit.test('pipeline dynamically configures itself based on input', function() {
 QUnit.test('reuses audio track object when the pipeline reconfigures itself', function() {
   var boxes, segments = [],
     id3Tag = new Uint8Array(73),
-    streamTimestamp = "com.apple.streaming.transportStreamTimestamp",
+    streamTimestamp = 'com.apple.streaming.transportStreamTimestamp',
     priv = 'PRIV',
-    i;
+    i, adtsPayload;
```
Comment (gesinger) [inline test/transmuxer.test.js]: Newline

File: lib/tools/flv-inspector.js [hunk 6]
```diff
@@ -6,65 +6,26 @@ var
...
   },
-  hexStringList = function (data) {
+  hexStringList = function(data) {
     var arr = [], i;
-    /* jshint -W086 */
-    while(data.byteLength > 0) {
+
+    while (data.byteLength > 0) {
       i = 0;
-      switch(data.byteLength) {
```
Comment (gesinger) [inline lib/tools/flv-inspector.js]: This switch was crazy.

**Comment by gesinger** [comment]
Minor notes, but aside from those LGTM

File: test/transmuxer.test.js (hunk 4 above)
Comment (dmlap) [inline test/transmuxer.test.js]: Meant to just remove the extra space around the `=` but I messed up the indentation while I was at it. Fixing.

File: test/transmuxer.test.js (hunk 3 above)
Comment (dmlap) [inline test/transmuxer.test.js]: It's an assignment so I have to perform it outside the `var` declarations. We *really* should clean up the declarations in this file sometime, though.

**Pull request merged**